profiles/common/modules/features/ec_embedded_image/ec_embedded_image.behat.inc

```diff
@@ -86,34 +86,21 @@ class ECEmbeddedImageSubContext extends DrupalSubContextBase {
       ]
     );
 
-    $this->contentTypeWasSetUp = TRUE;
+    $this->testPermissionsSetUp = TRUE;
   }
 
   /**
-   * Removes the test_media_gallery content type and related permissions.
+   * Removes the permissions related to the test_media_gallery content type.
+   *
+   * @see createMediaGalleryContentType()
    *
    * @AfterScenario
    */
-  public function removeMediaGalleryContentType() {
-    if (!$this->contentTypeWasSetUp) {
+  public function resetMediaGalleryPermission() {
+    if (!$this->testPermissionsSetUp) {
       return;
     }
 
-    $this->contentTypeWasSetUp = FALSE;
```

Comment:
Now I understood what @voidtek ment.

If we run the method createMediaGalleryContentType one time $testPermissionsSetUp will be true in all subsequent AfterScenarios and method resetMediaGalleryPermission will not return in first test.

@gillesdeudon can you put back "$this->contentTypeWasSetUp = FALSE;"? It is important to be there.